iothub/service/tests/DirectMethod/DirectMethodClientResponseTests.cs

```diff
@@ -175,13 +184,43 @@ public void DirectMethodClientResponse_Payload_CustomType()
             DirectMethodClientResponse dmcr = JsonConvert.DeserializeObject<DirectMethodClientResponse>(body);
 
             // assert
-
             dmcr.Status.Should().Be(expectedStatus);
             dmcr.TryGetPayload(out CustomType actual).Should().BeTrue();
             actual.CustomInt.Should().Be(expectedPayload.CustomInt);
             actual.CustomString.Should().Be(expectedPayload.CustomString);
         }
 
+        [TestMethod]
+        public void DirectMethodClientResponse_Payload_Null()
+        {
+            //arrange
+            const int expectedStatus = 200;
+            var source = new DirectMethodClientResponse
+            {
+                Status = expectedStatus,
+                JsonPayload = null,
+            };
+
+            // act and assert
+            source.TryGetPayload(out string _).Should().BeFalse();
+        }
+
+        [TestMethod]
+        public void DirectMethodClientResponse_Payload_ThrowsException()
+        {
+            //arrange
+            const int expectedStatus = 200;
+            var source = new DirectMethodClientResponse
+            {
+                Status = expectedStatus,
+                JsonPayload = new JRaw(JsonConvert.SerializeObject(TimeSpan.FromSeconds(30)))
+            };
+
+            // act and assert
+            // deliberately throw serialzation exception
```

Comment:
What does this mean? It shouldn't throw?

Reply:
I guess you mean it will fail (not throw). I'm not sure why it is expected to fail, though.